lib/membrane/core/child/pad_model.ex

```diff
@@ -84,16 +81,26 @@ defmodule Membrane.Core.Child.PadModel do
         if match?(unquote(pattern), data) do
           :ok
         else
-          {:error,
-           {:invalid_pad_data, ref: unquote(pad_ref), pattern: unquote(pattern), data: data}}
+          {:error, :no_match}
         end
       end
     end
   end
 
   defmacro assert_data!(state, pad_ref, pattern) do
     quote do
-      :ok = unquote(__MODULE__).assert_data(unquote(state), unquote(pad_ref), unquote(pattern))
+      pad_ref = unquote(pad_ref)
+      state = unquote(state)
+      data = unquote(__MODULE__).get_data!(state, pad_ref)
+
+      if match?(unquote(pattern), data) do
+        :ok
+      else
+        raise Membrane.PadError, """
+        Assertion on data of the pad #{inspect(pad_ref)} failed, pattern: #{unquote(Macro.to_string(pattern))}
+        Pad data: #{inspect(data, pretty: true)}
+        """
+      end
```

Comment:
I think case would be more idiomatic here (make sure to format this suggestion):
```suggestion
      case data do
        unquote(pattern) ->
          :ok
        _ ->
          raise Membrane.PadError, """
          Assertion on data of the pad #{inspect(pad_ref)} failed, pattern: #{unquote(Macro.to_string(pattern))}
          Pad data: #{inspect(data, pretty: true)}
          """
      end
```